.github/workflows/generate-summary.yaml

```diff
@@ -564,15 +579,34 @@ jobs:
       # These three conditions should be equivalent, however, add them just in case
       # we change something accidentally which breaks it
       - name: Add Valid-Baseline Label to New Issue
-        uses: actions/github-script@v6
         if: ${{ github.workflow == 'Run-Frequent' && inputs.prefix == '' && inputs.issue_hash_prefix == 'Testsuite Status' && steps.issue-labels.outputs.valid_baseline != 'false' }}
+        uses: ./.github/actions/common/save-label
+        with:
+          label: "'valid-baseline'"
+          output: ${{ steps.label-aggregation-preset.outputs.label_file }}
+
+      # Execute this step even if there was a previous failure. Labels must always be uploaded to the issue
+      - name: Labels upload setup
+        id: labels-upload-setup
+        if: ${{ success() || failure() }}
+        run: |
+          if [[ -e "${{ steps.label-aggregation-preset.outputs.label_file }}" && -s "${{ steps.label-aggregation-preset.outputs.label_file }}" ]]; then
+            echo "labels_created=true" >> "$GITHUB_OUTPUT"
+            echo "aggregated_labels=$(cat ${{ steps.label-aggregation-preset.outputs.label_file }})" >> "$GITHUB_OUTPUT"
+          else
+            echo "labels_created=false" >> "$GITHUB_OUTPUT"
+          fi
+
+      - name: Add all the labels to New Issue
+        uses: actions/github-script@v6
+        if: ${{ (success() || failure()) && steps.labels-upload-setup.outputs.labels_created == 'true' }}
```

Comment:
nit: since this step depends on the previous, you don't need to have the `success() || failure()` check

Reply:
Oh I'll test on this one before making a change. Didn't realize that the current step's execution only depends on the previous step if the `if` is not specified

Reply:
that's not exactly the case. It's only the case when there hasn't been a signal telling the action to stop. I imagine there aren't any situations where we only hit the `failure() == true` section?

Reply:
This would be the same as the above case. Without `success() || failure()`, github automatically checks `success()` before evaluating the if statement where success would check if all the previous steps succeeded. If there is any failure, this step won't execute even if the previous step gets executed.

Reply:
same as above